pkg/build/api/v1beta3/conversion.go

```diff
@@ -336,7 +336,28 @@ func init() {
 		func(in *ImageChangeTrigger, out *newer.ImageChangeTrigger, s conversion.Scope) error {
 			out.LastTriggeredImageID = in.LastTriggeredImageID
 			return nil
-		})
+		},
+		func(in *newer.BuildTriggerPolicy, out *BuildTriggerPolicy, s conversion.Scope) error {
+			if err := s.DefaultConvert(in, out, conversion.DestFromSource); err != nil {
+				return err
+			}
+			switch in.Type {
```

Comment:
@bparees if this looks good I'll add other cases in this switch for GitHub and Generic.